tests/library/library_node_expand_test.py

```diff
@@ -0,0 +1,151 @@
+
+    # Test with valid implementation
+    result = state.expand_library_node(addnode, 'pure')
+    assert result == 'pure'
+
+
+def test_state_expand_library_node_errors():
+    """Test error handling in SDFGState.expand_library_node method"""
+    sdfg, state, addnode = create_test_sdfg()
+
+    # Test with invalid implementation
+    with pytest.raises(KeyError) as exc_info:
+        state.expand_library_node(addnode, 'nonexistent')
+    assert 'Unknown implementation for node AddNode: nonexistent' in str(exc_info.value)
+
+    # Test with node not in state
+    sdfg2, state2, addnode2 = create_test_sdfg()
+    with pytest.raises(ValueError) as exc_info:
+        state.expand_library_node(addnode2, 'pure')
+    assert 'is not in this state' in str(exc_info.value)
+
+
+def test_library_node_expand_with_kwargs():
+    """Test that expansion kwargs are properly passed through"""
+    sdfg, state, addnode = create_test_sdfg()
+
+    # Test that the method accepts kwargs even if they aren't used
+    # We'll test with a library node that doesn't use kwargs but doesn't fail
+    try:
+        result = addnode.expand(state, 'pure')
+        assert result == 'pure'
+    except TypeError:
+        pytest.fail("expand method should handle kwargs gracefully")
+
+    # Test with state method - this should also work
+    sdfg2, state2, addnode2 = create_test_sdfg()
+    result2 = state2.expand_library_node(addnode2, 'pure')
+    assert result2 == 'pure'
+
+
+def test_compatibility_with_existing_expand_library_nodes():
+    """Test that the original SDFG.expand_library_nodes() still works"""
+    sdfg, state, addnode = create_test_sdfg()
+
+    # This should not raise any errors
+    sdfg.expand_library_nodes()
+
+    # The node should be expanded and no longer be a library node
+    # (it should be replaced by the expansion)
+    library_nodes = [n for n in state.nodes() if isinstance(n, dace.nodes.LibraryNode)]
+    assert len(library_nodes) == 0  # Should be expanded
+
+
+def test_functional_correctness():
+    """Test that the expanded node actually works correctly"""
+    sdfg, state, addnode = create_test_sdfg()
+
+    # Expand using new interface
+    addnode.expand(state, 'pure')
+
+    # Test execution
+    A = np.array([5.0], dtype=np.float32)
+    B = np.array([0.0], dtype=np.float32)
+    sdfg(A=A, B=B)
+
+    # AddNode should add 1 to the input (as per the implementation)
+    assert B[0] == 6.0
+
+
+def test_implementation_override():
+    """Test that the implementation parameter overrides the node's implementation"""
+    sdfg, state, addnode = create_test_sdfg()
+
+    # Initially, the node has no implementation set
+    assert addnode.implementation is None
+
+    # Set a different implementation on the node
+    addnode.implementation = 'nonexistent'  # This would normally fail
+
+    # But we override it with a valid implementation
+    result = addnode.expand(state, 'pure')
+    assert result == 'pure'
+
```

Comment:
missing test with expansion that has arguments/keyword arguments. For example, the Gemm library node from `dace/libraries/blas/nodes/gemm.py`  with alpha and beta values.

Reply:
Sorry, I didn't mean that. What I meant was a library node expansion where the `def expansion` static method has keyword arguments (just like we now specify in `expand`). Please make a test library node expansion with an expansion keyword argument you can test for in the resulting SDFG. I recommend adding a "Tiled" expansion for the add library node you use in the rest of the tests, with a tile_size keyword argument in `def expansion`. Use the decorator method (`dace/libraries/blas/nodes/einsum.py`, line 33) to register such a new test expansion only for this test. Check the expanded graph for the correct tile size.

You may also remove `test_gemm_expansion_with_arguments` because it is not relevant.